diff --git a/mesonbuild/mesonlib.py b/mesonbuild/mesonlib.py
index 59d4f81..ba58787 100644
--- a/mesonbuild/mesonlib.py
+++ b/mesonbuild/mesonlib.py
@@ -56,7 +56,8 @@ def set_meson_command(mainfile):
     global meson_command
     # On UNIX-like systems `meson` is a Python script
     # On Windows `meson` and `meson.exe` are wrapper exes
-    if not mainfile.endswith('.py'):
+    # ^^^ this is not true on Conda: for us, on Windows, `meson` is a wrapper script
+    if not mainfile.endswith('.py') and not sys.platform.startswith('win'):
         meson_command = [mainfile]
     elif os.path.isabs(mainfile) and mainfile.endswith('mesonmain.py'):
         # Can't actually run meson with an absolute path to mesonmain.py, it must be run as -m mesonbuild.mesonmain
