From 35f010089df188a992c1e994905d130a3e259aff Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 21 Sep 2023 23:49:37 +1100
Subject: [PATCH 2/4] minimal adaptations for llvm-flang

---
 mesonbuild/compilers/fortran.py | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/mesonbuild/compilers/fortran.py b/mesonbuild/compilers/fortran.py
index 7c836b74a..9915f6912 100644
--- a/mesonbuild/compilers/fortran.py
+++ b/mesonbuild/compilers/fortran.py
@@ -459,13 +459,18 @@ class FlangFortranCompiler(ClangCompiler, FortranCompiler):
                                  is_cross, info, exe_wrapper, linker=linker,
                                  full_version=full_version)
         ClangCompiler.__init__(self, {})
-        default_warn_args = ['-Minform=inform']
+        default_warn_args = []
         self.warn_args = {'0': [],
                           '1': default_warn_args,
                           '2': default_warn_args,
                           '3': default_warn_args,
                           'everything': default_warn_args}
 
+    # llvm-flang doesn't support -module (even though classic flang did), see
+    # https://github.com/llvm/llvm-project/issues/66969
+    def get_module_outdir_args(self, path: str) -> T.List[str]:
+        return ['-module-dir', path]
+
     def language_stdlib_only_link_flags(self, env: 'Environment') -> T.List[str]:
         # We need to apply the search prefix here, as these link arguments may
         # be passed to a different compiler with a different set of default
@@ -475,7 +480,7 @@ class FlangFortranCompiler(ClangCompiler, FortranCompiler):
         search_dirs: T.List[str] = []
         for d in self.get_compiler_dirs(env, 'libraries'):
             search_dirs.append(f'-L{d}')
-        return search_dirs + ['-lflang', '-lpgmath']
+        return search_dirs + ['-lFortranRuntime', '-lFortranDecimal']
 
 class ArmLtdFlangFortranCompiler(FlangFortranCompiler):
 
