From 74ab29035cf146bbfc48bcf747910ca337c27522 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 25 Sep 2023 09:19:10 +1100
Subject: [PATCH 3/4] mark lld as not supporting rsp

---
 mesonbuild/linkers/linkers.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mesonbuild/linkers/linkers.py b/mesonbuild/linkers/linkers.py
index edd408572..5b2954002 100644
--- a/mesonbuild/linkers/linkers.py
+++ b/mesonbuild/linkers/linkers.py
@@ -891,6 +891,9 @@ class LLVMDynamicLinker(GnuLikeDynamicLinkerMixin, PosixDynamicLinkerMixin, Dyna
     def get_thinlto_cache_args(self, path: str) -> T.List[str]:
         return ['-Wl,--thinlto-cache-dir=' + path]
 
+    def can_linker_accept_rsp(self) -> bool:
+        return False
+
     def get_win_subsystem_args(self, value: str) -> T.List[str]:
         # lld does not support a numeric subsystem value
         version = None
@@ -1339,6 +1342,9 @@ class ClangClDynamicLinker(VisualStudioLikeLinkerMixin, DynamicLinker):
 
         return super().get_output_args(outputname)
 
+    def can_linker_accept_rsp(self) -> bool:
+        return False
+
     def get_gui_app_args(self, value: bool) -> T.List[str]:
         return self.get_win_subsystem_args("windows" if value else "console")
 
