From aef3092b6a672dfa3b922cbb7aa705a40c4c2a4f Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 16 May 2024 15:58:16 +1100
Subject: [PATCH 3/3] override flags not known by flang

---
 mesonbuild/compilers/fortran.py | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/mesonbuild/compilers/fortran.py b/mesonbuild/compilers/fortran.py
index b9185efbe..d9471830d 100644
--- a/mesonbuild/compilers/fortran.py
+++ b/mesonbuild/compilers/fortran.py
@@ -13,7 +13,7 @@ from .compilers import (
     CompileCheckMode,
 )
 from .mixins.clike import CLikeCompiler
-from .mixins.gnu import GnuCompiler,  gnu_optimization_args
+from .mixins.gnu import GnuCompiler,  gnu_optimization_args, gnu_symbol_visibility_args
 from .mixins.intel import IntelGnuLikeCompiler, IntelVisualStudioLikeCompiler
 from .mixins.clang import ClangCompiler
 from .mixins.elbrus import ElbrusCompiler
@@ -447,6 +447,18 @@ class FlangFortranCompiler(ClangCompiler, FortranCompiler):
                           '3': default_warn_args,
                           'everything': default_warn_args}
 
+    def get_colorout_args(self, colortype: str) -> T.List[str]:
+        # clang's -fdiagnostics-color=... isn't known by flang (as of v18)
+        return []
+
+    def get_dependency_gen_args(self, outtarget: str, outfile: str) -> T.List[str]:
+        return []
+
+    def gnu_symbol_visibility_args(self, vistype: str) -> T.List[str]:
+        # flang doesn't support symbol visibility flag yet?
+        # https://github.com/llvm/llvm-project/issues/92459
+        return []
+
     # llvm-flang doesn't support -module (even though classic flang did), see
     # https://github.com/llvm/llvm-project/issues/66969
     def get_module_outdir_args(self, path: str) -> T.List[str]:
