From ad1cdd319ed939c6af41607e3ee6647f7d45a119 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 29 Sep 2023 00:45:58 +1100
Subject: [PATCH 4/4] override FlangCompiler.use_linker_args

---
 mesonbuild/compilers/fortran.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mesonbuild/compilers/fortran.py b/mesonbuild/compilers/fortran.py
index 9915f6912..d4884f51b 100644
--- a/mesonbuild/compilers/fortran.py
+++ b/mesonbuild/compilers/fortran.py
@@ -471,6 +471,12 @@ class FlangFortranCompiler(ClangCompiler, FortranCompiler):
     def get_module_outdir_args(self, path: str) -> T.List[str]:
         return ['-module-dir', path]
 
+    @classmethod
+    def use_linker_args(cls, linker: str, version: str) -> T.List[str]:
+        # ensure we don't fall through to ClangCompiler.use_linker_args
+        # --> GnuLikeCompiler.use_linker_args, which doesn't support lld-link
+        return [f'-fuse-ld={linker}']
+
     def language_stdlib_only_link_flags(self, env: 'Environment') -> T.List[str]:
         # We need to apply the search prefix here, as these link arguments may
         # be passed to a different compiler with a different set of default
