From 198466b45f3cce987f615f97a082f84a58df5f83 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 17 May 2024 10:33:54 +1100
Subject: [PATCH 4/4] explicitly link Fortran_main

---
 mesonbuild/compilers/fortran.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/mesonbuild/compilers/fortran.py b/mesonbuild/compilers/fortran.py
index d9471830d..eec42c07d 100644
--- a/mesonbuild/compilers/fortran.py
+++ b/mesonbuild/compilers/fortran.py
@@ -473,7 +473,11 @@ class FlangFortranCompiler(ClangCompiler, FortranCompiler):
         search_dirs: T.List[str] = []
         for d in self.get_compiler_dirs(env, 'libraries'):
             search_dirs.append(f'-L{d}')
-        return search_dirs + ['-lFortranRuntime', '-lFortranDecimal']
+        # explicitly link to Fortran_main due to
+        # https://github.com/llvm/llvm-project/commit/9d6837d595719904720e5ff68ec1f1a2665bdc2f
+        # note that this changed again in flang 19 with
+        # https://github.com/llvm/llvm-project/commit/8d5386669ed63548daf1bee415596582d6d78d7d
+        return search_dirs + ['-lFortranRuntime', '-lFortranDecimal', '--whole-archive', '-lFortran_main']
 
 class ArmLtdFlangFortranCompiler(FlangFortranCompiler):
 
